har_dataset.py

```diff
@@ -825,23 +898,24 @@ def __prepare_sisfall__(path, split):
     recordings = recordings.with_columns(
         pl.col('class_name').map_dict(activities_map).alias('class').cast(pl.Int32)
     )
-
-    dfs_by_split = []
-    for group_name, data in recordings.group_by(__get_separating_cols__(dataset_name='sisfall'), maintain_order=True):
-        # subselect split first 70% for train, next 15% for val, next 15% for test
-        total_rows = data.shape[0]
-        val_start_row = round(total_rows * 0.7)
-        test_start_row = round(total_rows * 0.85)
-
-        match split:
-            case 'train':
-                dfs_by_split.append(data[0:val_start_row])
-            case 'val':
-                dfs_by_split.append(data[val_start_row:test_start_row])
-            case 'test':
-                dfs_by_split.append(data[test_start_row:])
-
-    recordings = pl.concat(dfs_by_split, how='vertical')
+    
+    if half_data==False:
```

Comment:
I think this if is not necessary, the group_by should be called always, as the subjects are skipped in the beginning of the loop